logL_SCSH.m

```diff
@@ -146,14 +146,14 @@
             % Compute derivative for dynamic parameters, scalings, offsets and
             % sigma_noise parameters
             dlogL_Cdy = reshape(-J_D_C.dY, size(Data{s}.SCSH.C));
-            dlogL_Cdxi = squeeze(nansum(nansum(dCydxi .* repmat(dlogL_Cdy, [1 1 size(dmydxi, 3)]), 2), 1));
+            dlogL_Cdxi = squeeze(nansum(nansum(nansum(dCydxi .* repmat(dlogL_Cdy, [1 1 1 size(dCydxi, 4)]), 2), 3), 1));
 
```

Comment:
Why do we need this component in an additional dimension? Do we need it generally and does it affect the outcome for the gradient? Or is it just something chainrule-ish related and does only apply for the log-case?

Reply:
Because of the chain rule thing. `dCydxi` is [1 * 5 * 5 * 31], `dlogL_Cdy` is [1 * 5 * 5]. So if we don't add this additional dimension, it will simply not work because dimension does not match. But I think `bsxfun(@times, ...)` will also work without this additional dimension if you prefer.